### PR TITLE
Updated failure event to include error code

### PR DESCRIPTION
**Pull request by laurence-bird**

### CODE REVIEW

**Comment by cb372** [comment]
It would be nice to produce the more detailed error codes (`ProfileRetrievalFailed`, `InvalidProfile`) as well, but I think that would require changes to the profile service. I think this is fine for now.

File: src/test/scala/com/ovoenergy/orchestration/ServiceTestIT.scala [hunk 1]
```diff
@@ -69,39 +69,40 @@ class ServiceTestIT extends FlatSpec
...
+     _ =>
+       val failures = commFailedConsumer.poll(30000).records(failedTopic).asScala.toList
+       failures.size shouldBe 1
+       failures.foreach(record => {
+         val failure = record.value().getOrElse(fail("No record for ${record.key()}"))
+         failure.reason should include("Customer has no usable email address")
+         failure.reason should include("Customer has no last name")
+         failure.reason should include("Customer has no first name")
+         failure.metadata.traceToken shouldBe TestUtil.traceToken
+       })
```
Comment (fleetwoods) [inline src/test/scala/com/ovoenergy/orchestration/ServiceTestIT.scala]: Should this not be where we use the incomplete profile error code?

File: src/test/scala/com/ovoenergy/orchestration/ServiceTestIT.scala [hunk 2]
```diff
@@ -69,39 +69,40 @@ class ServiceTestIT extends FlatSpec
...
+   val future = triggeredProducer.send(new ProducerRecord[String, Triggered](triggeredTopic, TestUtil.triggered))
+   whenReady(future) {
+     _ =>
+       val failures = commFailedConsumer.poll(30000).records(failedTopic).asScala.toList
+       failures.size shouldBe 1
+       failures.foreach(record => {
+         val failure = record.value().getOrElse(fail("No record for ${record.key()}"))
+         failure.reason should include("Error response (500) from profile service: Some error")
+         failure.metadata.traceToken shouldBe TestUtil.traceToken
+         failure.errorCode shouldBe OrchestrationError
```
Comment (fleetwoods) [inline src/test/scala/com/ovoenergy/orchestration/ServiceTestIT.scala]: ProfileRetrievalFailed?

**Comment by ovo-comms-circleci** [comment]
This PR was successfully deployed to production by [CircleCI build 52](https://circleci.com/gh/ovotech/comms-orchestration/52) at 2017-01-10 15:10 GMT.

**Pull request merged**